modules/rostests/apitests/version/VerQueryValue.c

```diff
@@ -0,0 +1,302 @@
+    ok(result, "GetFileVersionInfoA failed\n");
+
+    /* Query available translations */
+    result = VerQueryValueA(pvVersionInfoA,
+                            "\\VarFileInfo\\Translation",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 4, "Unexpected value\n");
+    ok((cbLen & 0x3) == 0, "Unexpected value\n");
+    uLanguage = ((USHORT*)pvData)[0];
+    uCodePage = ((USHORT*)pvData)[1];
+
+    /* Query sublock */
+    sprintf(szSubBlock, "\\StringFileInfo\\%04X%04X\\CompanyName", uLanguage, uCodePage);
+    result = VerQueryValueA(pvVersionInfoA,
+                            szSubBlock,
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 2, "Unexpected value\n");
+    ok((cbLen & 0x1) == 0, "Unexpected value\n");
+
+    free(pvVersionInfoA);
+}
+
+void
+Test_VerQueryValueW(void)
+{
+    DWORD dwVersionInfoSizeW;
+    PVOID pvVersionInfoW, pvData = NULL;
+    WCHAR szSubBlock[256];
+    USHORT uLanguage, uCodePage;
+    UINT cbLen;
+    BOOL result;
+
+    /* Get the size for the unicode version info */
+    dwVersionInfoSizeW = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
+    ok(dwVersionInfoSizeW > 0, "GetFileVersionInfoSizeW failed\n");
+
+    /* Get the unicode version info from kernel32 */
+    pvVersionInfoW = malloc(dwVersionInfoSizeW);
+    result = GetFileVersionInfoW(L"kernel32.dll", 0, dwVersionInfoSizeW, pvVersionInfoW);
+    ok(result, "GetFileVersionInfoW failed\n");
+
+    /* Query available translations */
+    result = VerQueryValueW(pvVersionInfoW,
+                            L"\\VarFileInfo\\Translation",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 4, "Unexpected value\n");
+    ok((cbLen & 0x3) == 0, "Unexpected value\n");
+    uLanguage = ((USHORT*)pvData)[0];
+    uCodePage = ((USHORT*)pvData)[1];
+
+    /* Query sublock */
+    _swprintf(szSubBlock, L"\\StringFileInfo\\%04X%04X\\CompanyName", uLanguage, uCodePage);
+    result = VerQueryValueW(pvVersionInfoW,
+                            szSubBlock,
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 2, "Unexpected value\n");
+    ok((cbLen & 0x1) == 0, "Unexpected value\n");
+
+    free(pvVersionInfoW);
+}
+
+void
+Test_StaticVersionInfo(void)
+{
+    PVOID pVersionInfo;
+    PVOID pvData = NULL;
+    SIZE_T ExpectedOffset;
+    UINT cbLen;
+    BOOL result;
+
+    /* Make a copy of the version info. Windows actually writes to it!
+       We make the buffer twice as big to account for the ANSI copy,
+       even if we don't use it. */
+    pVersionInfo = malloc(2 * sizeof(g_VersionInfo));
+    memset(pVersionInfo, 0, 2 * sizeof(g_VersionInfo));
```

Comment:
or malloc + memset(..., 0, ...)  --> calloc ?

Reply:
I don't think that is an improvement